[17.0][FIX] sign_oca: list requests from recent sent to oldest signed

kobros-tech opened the pull request:
Add pre_migration script to be sure about the field update

from:
https://github.com/OCA/sign/pull/123

kobros-tech committed:
[FIX] sign_oca: list requests from recent sent to oldest signed

Add pre_migration script to be sure about the field update

kobros-tech commented:
2025-09-02 15:46:37,343 17524 DEBUG sign_oca_17_1 OpenUpgrade: 8 rows affected after 0:00:00.001123 running 
        UPDATE sign_oca_request
        SET state = CASE
            WHEN state = 'sent' THEN '0_sent'
            WHEN state = 'draft' THEN '1_draft'
            WHEN state = 'signed' THEN '2_signed'
            WHEN state = 'cancel' THEN '3_cancel'
            ELSE state
        END

kobros-tech commented:
@pedrobaeza 
your approval too, and we merge in 17.0

etobella commented:
Tests are failing @kobros-tech

kobros-tech commented:
@pedrobaeza 
@etobella 

I would recommend we improve the addons list in OCA project so that dependent modules do not interfer the tests running for their dependent ones like our case here now, I guess this will make workflow more efficient.

kobros-tech commented:
> Tests are failing @kobros-tech

This will enforce me to update other modules, we agreed on sign_oca only :)

"agreement_sign_oca"

pedrobaeza commented:
You have to modify `agreement_sign_oca` to adapt to the new state values.

kobros-tech commented:
@pedrobaeza 

Can I do this in the same commit and PR?

this current one?

etobella commented:
@kobros-tech actually, tests are done this way for this reason. Now we are ensuring that nothing is broken when we migrate. It is making fail a test. We might need to fix it. That's all...

pedrobaeza commented:
> Can I do this in the same commit and PR?
> 
> this current one?

Yes, it should be done in the same PR, but it can be a different commit.

kobros-tech commented:
@pedrobaeza 
@etobella 

I would like to update other module in this same PR, maybe different commit.

What do you think?

kobros-tech committed:
[FIX] agreement_sign_oca: update state values

Based on our update on sign.request.state values to be prefixed with
numbers so that we can sequence requests based on the state field.

kobros-tech commented:
@etobella 
@pedrobaeza 

Wow, any comments?

pedrobaeza reviewed (approved):
Now only 18 is pending:

/ocabot merge minor
